tests/app/main/views/test_suppliers.py

```diff
@@ -314,17 +319,46 @@ def test_should_show_correct_fields_for_disabled_service(self, data_api_client):
     @mock.patch('app.main.views.suppliers.data_api_client')
     def test_should_show_correct_fields_for_enabled_service(self, data_api_client):
         data_api_client.get_supplier.return_value = self.load_example_listing("supplier_response")
+        data_api_client.find_services.return_value = self.load_example_listing("services_response")
+        data_api_client.find_frameworks.return_value = {
+            'frameworks': [self.load_example_listing("framework_response")['frameworks']]
+        }
 
-        services = self.load_example_listing("services_response")
-        services["services"][0]["status"] = "enabled"
-        data_api_client.find_services.return_value = services
+        data_api_client.find_services.return_value["services"][0]["status"] = "enabled"
 
         response = self.client.get('/admin/suppliers/services?supplier_id=1000')
 
         assert response.status_code == 200
         assert "Private" in response.get_data(as_text=True)
         assert "Edit" in response.get_data(as_text=True)
 
+    @mock.patch('app.main.views.suppliers.data_api_client')
+    def test_should_show_separate_tables_for_frameworks_if_supplier_has_service_on_framework(self, data_api_client):
+        data_api_client.get_supplier.return_value = self.load_example_listing("supplier_response")
+
+        service_1 = self.load_example_listing("services_response")['services'][0]
+        service_2 = service_1.copy()
+        service_2['frameworkName'] = 'Digital Outcomes and Specialists 2'
+        service_2['frameworkSlug'] = 'digital-outcomes-and-specialists-2'
+        data_api_client.find_services.return_value = {'services': [service_1, service_2]}
+
+        framework_1 = self.load_example_listing("framework_response")['frameworks']
+        framework_2 = framework_1.copy()
+        framework_2['name'] = 'Digital Outcomes and Specialists 2'
+        framework_2['slug'] = 'digital-outcomes-and-specialists-2'
+        framework_2['id'] = 5
+        data_api_client.find_frameworks.return_value = {'frameworks': [framework_1, framework_2]}
+
+        response = self.client.get('/admin/suppliers/services?supplier_id=1000')
+
+        assert response.status_code == 200
+        response_data = response.get_data(as_text=True)
+        assert 'g-cloud-8_services' in response_data
+        assert 'digital-outcomes-and-specialists-2_services' in response_data
+        gcloud_table_index = response_data.find('g-cloud-8_services')
+        dos_table_index = response_data.find('digital-outcomes-and-specialists-2_services')
+        assert gcloud_table_index < dos_table_index
```

Comment:
No xpath! 🙀 (This is fine, I think, although do we care what order the frameworks are listed? Is it meant to be most recent first? Or 'live' before 'expired'?)

Reply:
Yeah, in the screenshot G-Cloud and DOS frameworks are interleaved, I *think* by age?  Maybe a test with more than two frameworks would be good.

Reply:
Hey, the screenshot is split. I can add another framework to that test though

Reply:
Done